### DISCUSSION TITLE
New Feature Suggestions and Ideas

### DISCUSSION THREAD

**Discussion by sachinkar** [pranjal-joshi/Screeni-py#116] [Ideas]
Post all your ideas and suggestions for features/changes for Screeni-py here in this thread.

**Comment by sachinkar** [comment]
### Idea-1
As I've tested the tool soo far, I've found the "screenipy.ini" config file is quite useful and give screeni-py a lot of configurability. One that I think should be added is defining window/terminal size. As I can see every time we launch the terminal we can change and resize the terminal as per our view, but that change is temporary. So if we close that setting is not saved. I think we can add a "TerminalSize" option in the ini for defining and saving your most preferred window size.

**Reply by pranjal-joshi** [reply]
Screeni-py is designed to be a cross-platform tool and every OS handles the terminal fonts in a specific way.
If you're running on Windows, I recommend you to **use Windows Terminal instead of Command Prompt** as it supports the Unicode and user profiles which will save your font/size config for subsequent executions.

For Linux/MacOS, Feel free to know more about your desktop environment like `gnome` and how to configure it.

#### Windows Terminal - Custom Configuration
![image](https://user-images.githubusercontent.com/6128978/168133295-6608b7e3-08ac-4d4c-8219-3cdc1313efad.png)

**Reply by sachinkar** [reply]
Great info thanks.

**Comment by sachinkar** [comment]
There are some patterns that I noticed are missing such as Bullish/Bearish Harami, The Evening Star, NR7, Tweezer Top etc as we can study from this page(s): https://www.elearnmarkets.com/blog/35-candlestick-patterns-in-stock-market/ and https://www.elearnmarkets.com/blog/nr4-and-nr7-trading-strategy-setup/

Any plans to add these?

**Reply by pranjal-joshi** [reply]
Hi, **Narrow Range (x-days)** has been already implemented in earlier version (2666ecf9c9d892ae6e426a0aa81b34c9beca3198) and it's still supported.
Please go through the code if required.
https://github.com/pranjal-joshi/Screeni-py/blob/6d9f406cace8c8579fe04f7e3301de3ba7aa88b4/src/classes/Screener.py#L438-L462

Following Candlestick Patterns have been already implemented. Feel free to contribute for more by referring documentation: https://github.com/mrjbq7/ta-lib#pattern-recognition
https://github.com/pranjal-joshi/Screeni-py/blob/6d9f406cace8c8579fe04f7e3301de3ba7aa88b4/src/classes/CandlePatterns.py#L12-L15

**Comment by rohitnirantar** [comment]
Few features which will add great value add:
1. Getting list of stocks which are performing better than broader index/ Sector Index
2. No of Stocks making new high/Low on Daily/Weekly timeframe
3. Stocks closing above 200 DMA

**Reply by pranjal-joshi** [reply]
Thanks for your inputs @rohitnirantar 
* Please check the latest update where Stocks making new highs can be found using `Option > 2 (Breakout)` (Checkout how to edit `screenipy.ini` for better customizations about the timeframes!)
* Explore the option to find screening by MA reversals. Here you can find reversal at desired MA length or even MA confluence signals to get better trading opportunities.
